src/libexpr/nixexpr.cc

```diff
@@ -581,6 +581,20 @@ std::string ExprLambda::showNamePos(const EvalState & state) const
     return fmt("%1% at %2%", id, state.positions[pos]);
 }
 
+void ExprLambda::setDocComment(DocComment docComment) {
+    if (!this->docComment) {
```

Comment:
why `setDocComment` method cannot "reset" the comment if it already exists?

Reply:
Removing this would cause the outermost comment to take precedence over the closest one. This behavior conforms with [this part](https://github.com/NixOS/rfcs/pull/145/files#diff-5050f8573da95e8531bf14e0b0de89d58a680478678ed4f2ed20f9e53def4dd9R200-R207) of the RFC.